### ISSUE TITLE
Can't find a way to identify the user

### ISSUE THREAD

**Issue by grahamoptibrium** [O365/python-o365#324]
I have two use-cases:
- Allow a user to log in to our private application with their 365 credentials 
- Send automated emails on behalf of the user

I would like to use the same library for both.

Currently I can authenticate a user successfully by sending them off to microsoftonline, but then can't seem to identify that user from the account object, which makes it useless for both use-cases.

**Comment by janscas** [comment]
I don't understand.

If a user logins and gives consent to your app, and this consent includes permission to send emails, then your app can send emails as the user. So .. 

a) He is able to login into your app
b) Your app can send emails on behalf of the user

That's all you are asking for?

**Comment by grahamoptibrium** [comment]
@janscas the problem is I don't know who has logged into the app ...

**Comment by janscas** [comment]
Ok, this has been ask before.

This will be implemented sometime.

for now: See #310

**Comment by grahamoptibrium** [comment]
@janscas cool. It might prove to be a useful feature.

For anyone who finds this issue on Google: As an interim solution, the JWT contains the `unique_name` and `upn`(primary email) of the user. Although a horrible abomination against nature, I reliably got at it in the following way:

```python
from O365.utils import BaseTokenBackend
class ObjectBackend(BaseTokenBackend):

    def token(self, value):
        self.token = value

    def save_token(self):
        pass

    def get_token(self):
        return self.token['access_token']


backend = ObjectBackend()
account = Account(credentials, token_backend=backend)
authenticated = account.con.request_token(request.url, redirect_uri=my_callback)

if authenticated:
    token = backend.get_token().split('.')[1]
    # RFC7515 specifies removal of base64 padding from the JWT
    # So we pad it with '===' to make it a valid base64 string
    email = json.loads(base64.b64decode(token + '==='))['unique_name']
```

**Comment by janscas** [comment]
You can’t rely on this as the unique_name is only contained in the v1 auth token.

**Comment by janscas** [comment]
Now implemented on version 2.0.5
See 11560d0
